CDDSO-648: Unify cell_methods when reading ocean data by removing "interval" specifications

matthew-mizielinski opened the pull request:
Some UKESM models have experienced problems due to changes in ocean timestep being made mid year as this propagates into the cell_methods attribute attached to data variables in the model input files.

This change adds an extra step before cubes are concatenated together within MIP Convert where the cell_methods are checked and, if necessary, unified by removing the "intervals" component of the CellMethod.

Note: this change should not affect PP atmosphere data as the timestep is not encoded in the PP header.

mo-kerstinschmatzer reviewed (commented):
LGTM

But tests are failing because of code style issues.

matthew-mizielinski commented:
> LGTM
> 
> But tests are failing because of code style issues.

Thanks -- I've fixed these tests and amended the docstring for the new function.

mo-jareddrayton reviewed (changes requested):
A few docstring/naming things (still looking at code changes).

- The docstring return section can go as cubes are modified in place
- The docstring `cubes` argument should be a `iris.cube.CubeList` rather than `iris.cube.Cube`.
- The function says `remove_cell...` but as far as I can tell it is unifying the interval cell methods rather than removing them. Might be better to rename the function. EDIT: I see that they are removed.
- The docstring format like a mix of both NumPy and Google style. (not that it matters that much as there are already at least three distinct docstring styles used throughout `mip_convert`...)

matthew-mizielinski commented:
> A few docstring/naming things (still looking at code changes).
> 
> * The docstring return section can go as cubes are modified in place
> * The docstring `cubes` argument should be a `iris.cube.CubeList` rather than `iris.cube.Cube`.
> * The function says `remove_cell...` but as far as I can tell it is unifying the interval cell methods rather than removing them. Might be better to rename the function. EDIT: I see that they are removed.
> * The docstring format like a mix of both NumPy and Google style. (not that it matters that much as there are already at least three distinct docstring styles used throughout `mip_convert`...)

Thanks for catching this Jared, I think I started the docstring when I had accidentally put the code within one of the iris modules (don't ask) and failed to correct it when I moved the function to its current home.

Docstring simplified and updated format in 51c74cba

matthew-mizielinski commented:
cherry picked to main in 42c182f6
